lib/src/mamac_device_commands.dart

```diff
@@ -72,3 +77,79 @@ class EditDevice extends SimpleNode {
     (parent as MamacDeviceNode).update(params);
   }
 }
+
+class GetLogsNode extends SimpleNode {
+  static const String isType = 'getLogsNode';
+  static const String pathName = 'Get_Logs';
+  static const String childNodePrefix = 'Logs_';
+
+  static Map<String, dynamic> definition() => {
+        r'$is': isType,
+        r'$name': 'Get Logs',
+        r'$invokable': 'write',
+        r'$params': [],
+        r'$columns': [
+          {'name': 'success', 'type': 'bool', 'default': false},
+          {'name': 'message', 'type': 'string', 'default': ''}
+        ]
+      };
+
+  MamacDevice _device;
+  GetLogsService _getLogsService = new GetLogsService();
+  LinkProvider _link;
+
+  void set device(MamacDevice device) {
```

Comment:
I think rather than needing to do this every time, just pass the device to the definition. That way `params` can be set before it's initialized rather than setting it and changing it after the fact.